### PR TITLE
convert python script into javascript

### PR DESCRIPTION
**Pull request by webelf101**
fixes https://github.com/VolumeFi/paloma/issues/156

### CODE REVIEW

File: execute.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+// amount0: uint256
+// amount1_min: uint256
+// amount1_max: uint256
+// depositor: address
+
+let db = new sqlite3.Database("./events.db");
+db.serialize(() => {
+    let sql = `CREATE TABLE IF NOT EXISTS fetched_blocks (ID INTEGER PRIMARY KEY AUTOINCREMENT, block_number INTEGER);`;
+    db.run(sql);
+    sql = `CREATE TABLE IF NOT EXISTS deposits (ID INTEGER PRIMARY KEY AUTOINCREMENT, deposit_id INTEGER, token0 TEXT, token1 TEXT, amount0 TEXT, amount1_min TEXT, amount1_max TEXT, pool TEXT, depositor TEXT, profit_taking_or_stop_loss NUMERIC, withdraw_block INTEGER, withdrawer TEXT);`;
```
[the remark this answers is not in the document]
Comment (webelf101) [reply execute.js]: fixed

File: execute.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                flat_array.push(deposited_events[key].returnValues["amount0"]);
+                flat_array.push(deposited_events[key].returnValues["amount1_min"]);
+                flat_array.push(deposited_events[key].returnValues["amount1_max"]);
+                flat_array.push(deposited_events[key].returnValues["pool"]);
+                flat_array.push(deposited_events[key].returnValues["depositor"]);
+            }
+            db.run(sql, flat_array);
+        }
+        if (withdrawn_events.length != 0) {
+            let sql = `UPDATE deposits SET withdraw_block = ?, withdrawer = ?, profit_taking_or_stop_loss = ? WHERE deposit_id = ?;`;
```
[the remark this answers is not in the document]
Comment (webelf101) [reply execute.js:116]: fixed.

File: execute.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            token0 TEXT NOT NULL,
+            token1 TEXT NOT NULL,
+            amount0 TEXT NOT NULL,
+            amount1_min TEXT NOT NULL,
+            amount1_max TEXT NOT NULL,
+            pool TEXT NOT NULL,
+            depositor TEXT NOT NULL,
+            profit_taking_or_stop_loss NUMERIC,
+            withdraw_block INTEGER,
+            withdrawer TEXT
```
Comment (measure-fi) [inline execute.js:62]: Also `NOT NULL`?

File: execute.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    db.run(
+        `CREATE TABLE IF NOT EXISTS fetched_blocks (
+            ID INTEGER PRIMARY KEY AUTOINCREMENT,
+            block_number INTEGER
+        );`
+    );
+    db.run(
+        `CREATE TABLE IF NOT EXISTS deposits (
+            ID INTEGER PRIMARY KEY AUTOINCREMENT,
+            deposit_id INTEGER NOT NULL,
```
Comment (measure-fi) [inline execute.js:52]: Why are these two columns instead of one column?

File: execute.js (hunk 3 above)
Comment (webelf101) [reply execute.js:62]: They are NULL.

File: execute.js (hunk 4 above)
Comment (webelf101) [reply execute.js:52]: deposit id comes from events and in some reasons in chain issue, the deposit_id and ID might not be the same.

Comment (measure-fi) [reply execute.js:52]: Can we just make them the same or is there a reason deposit_id's would ever be non unique? I guess it doesn't really matter if this is what you want.

File: execute.js (hunk 3 above)
Comment (measure-fi) [reply execute.js:62]: Now I'm even more confused, what does it mean for `profit_taking_or_stop_loss` to be true, what does it mean for it to be false, and what does it mean for it to be null?

Comment (webelf101) [reply execute.js:62]: NULL: not withdrawn yet(pending for withdrawing).
true: withdrawn as a profit taking.
false: withdrawn as a stop loss.

File: execute.js (hunk 4 above)
Comment (webelf101) [reply execute.js]: I removed the ID. Are you satisfied? 😉

Comment (measure-fi) [reply execute.js]: Never!

**Review by measure-fi — APPROVED** [review]
